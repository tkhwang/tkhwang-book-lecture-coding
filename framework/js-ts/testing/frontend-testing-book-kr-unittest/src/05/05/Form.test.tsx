import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Form } from "./Form";

const user = userEvent.setup();

test("form의 접근 가능한 이름은 heading에서 인용합니다", () => {
  render(<Form />);
  expect(
    screen.getByRole("form", { name: "신규 계정 등록" })
  ).toBeInTheDocument();
});

test("주요 영역이 표시되어 있다", () => {
  render(<Form />);
  expect(
    screen.getByRole("heading", { name: "신규 계정 등록" })
  ).toBeInTheDocument();
  expect(
    screen.getByRole("group", { name: "계정정보 입력" })
  ).toBeInTheDocument();
  expect(
    screen.getByRole("group", { name: "이용 약관 동의" })
  ).toBeInTheDocument();
  expect(
    screen.getByRole("button", { name: "회원가입" })
  ).toBeInTheDocument();
});

test("회원가입 버튼은 비활성화 상태다", () => {
  render(<Form />);
  expect(screen.getByRole("button", { name: "회원가입" })).toBeDisabled();
});

test("이용 약관에 동의하는 체크 박스를 클릭하면 회원가입 버튼은 활성화된다", async () => {
  render(<Form />);
  await user.click(screen.getByRole("checkbox"));
  expect(screen.getByRole("button", { name: "회원가입" })).toBeEnabled();
});

test("Snapshot: 신규 계정 등록 폼이 표시된다", () => {
  const { container } = render(<Form />);
  expect(container).toMatchSnapshot();
});
