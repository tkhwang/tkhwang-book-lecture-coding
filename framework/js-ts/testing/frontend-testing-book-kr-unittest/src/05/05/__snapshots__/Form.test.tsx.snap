// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: 신규 계정 등록 폼이 표시된다 1`] = `
<div>
  <form
    aria-labelledby=":r4:"
  >
    <h2
      id=":r4:"
    >
      신규 계정 등록
    </h2>
    <fieldset>
      <legend>
        계정정보 입력
      </legend>
      <div>
        <label>
          메일주소
          <input
            placeholder="example@test.com"
            type="text"
          />
        </label>
      </div>
      <div>
        <label>
          비밀번호
          <input
            placeholder="8자 이상"
            type="password"
          />
        </label>
      </div>
    </fieldset>
    <fieldset>
      <legend>
        이용 약관 동의
      </legend>
      <label>
        <input
          type="checkbox"
        />
        서비스 
        <a
          href="/terms"
        >
          이용 약관
        </a>
        을 확인했으며 이에 동의합니다
      </label>
    </fieldset>
    <div>
      <button
        disabled=""
      >
        회원가입
      </button>
    </div>
  </form>
</div>
`;
